pkg/clusteragent/externalmetrics/model/datadogmetricinternal.go

```diff
@@ -106,9 +109,30 @@ func NewDatadogMetricInternalFromExternalMetric(id, query, metricName, autoscale
 	}
 }
 
-// UpdateFrom updates the `DatadogMetricInternal` from `DatadogMetric` Spec, returns modified instance
+// GetQuery returns the query that should be used to fetch metrics
+func (d *DatadogMetricInternal) GetQuery() string {
```

Comment:
I'd maybe simplify to `Query` and `RawQuery` (to be used for autogen, even though it should work with `Query`)

Reply:
TBH I preferred `GetQuery` because `Query` can be interpreted as a verb (very misleading in this context) 😅  no strong opinion though, I can change it